PeakCoreData/Core/Protocols/ManagedObjectType.swift

```diff
@@ -32,7 +32,7 @@ public extension ManagedObjectType {
      */
     @discardableResult
     static func insertObject(in context: NSManagedObjectContext, configure: ManagedObjectConfigurationBlock? = nil) -> Self {
-        let object = Self(context: context)
+        guard let object = NSEntityDescription.insertNewObject(forEntityName: entityName, into: context) as? Self else { fatalError("Wrong object type inserted.")}
```

Comment:
Why not just do `as!` instead of `as?` then `fatalError`? Seems redundant to me.

Reply:
if we use ! the app just crashes and we have no idea why. If we use fatalError we get to print out the reason why the app crashed before it crashes.

Reply:
Ehhh sure but I think you should only use ! where you are certain as a developer it won't be nil. But I will concede the point to you.